contracts/src/v0.8/vrf/dev/VRFCoordinatorV2_5.sol

```diff
@@ -379,123 +390,140 @@ contract VRFCoordinatorV2_5 is VRF, SubscriptionAPI, IVRFCoordinatorV2Plus {
    * @return payment amount billed to the subscription
    * @dev simulated offchain to determine if sufficient balance is present to fulfill the request
    */
-  function fulfillRandomWords(Proof memory proof, RequestCommitment memory rc) external nonReentrant returns (uint96) {
+  function fulfillRandomWords(
+    Proof memory proof,
+    RequestCommitment memory rc,
+    bool onlyPremium
+  ) external nonReentrant returns (uint96) {
     uint256 startGas = gasleft();
     Output memory output = _getRandomnessFromProof(proof, rc);
+    uint256 gasPrice = _getValidatedGasPrice(onlyPremium, output.provingKey.maxGas);
 
     uint256[] memory randomWords = new uint256[](rc.numWords);
     for (uint256 i = 0; i < rc.numWords; i++) {
       randomWords[i] = uint256(keccak256(abi.encode(output.randomness, i)));
     }
 
     delete s_requestCommitments[output.requestId];
-    VRFConsumerBaseV2Plus v;
-    bytes memory resp = abi.encodeWithSelector(v.rawFulfillRandomWords.selector, output.requestId, randomWords);
-    // Call with explicitly the amount of callback gas requested
-    // Important to not let them exhaust the gas budget and avoid oracle payment.
-    // Do not allow any non-view/non-pure coordinator functions to be called
-    // during the consumers callback code via reentrancyLock.
-    // Note that _callWithExactGas will revert if we do not have sufficient gas
-    // to give the callee their requested amount.
-    s_config.reentrancyLock = true;
-    bool success = _callWithExactGas(rc.callbackGasLimit, rc.sender, resp);
-    s_config.reentrancyLock = false;
+    bool success = _deliverRandomness(output.requestId, rc, randomWords);
 
     // Increment the req count for the subscription.
-    uint64 reqCount = s_subscriptions[rc.subId].reqCount;
-    s_subscriptions[rc.subId].reqCount = reqCount + 1;
-
     // stack too deep error
     {
-      bool nativePayment = uint8(rc.extraArgs[rc.extraArgs.length - 1]) == 1;
-      // We want to charge users exactly for how much gas they use in their callback.
-      // The gasAfterPaymentCalculation is meant to cover these additional operations where we
-      // decrement the subscription balance and increment the withdrawable balance.
-      uint96 payment = _calculatePaymentAmount(
-        startGas,
-        s_config.gasAfterPaymentCalculation,
-        tx.gasprice,
-        nativePayment
-      );
-      if (nativePayment) {
-        if (s_subscriptions[rc.subId].nativeBalance < payment) {
-          revert InsufficientBalance();
-        }
-        s_subscriptions[rc.subId].nativeBalance -= payment;
-        s_withdrawableNative += payment;
-      } else {
-        if (s_subscriptions[rc.subId].balance < payment) {
-          revert InsufficientBalance();
-        }
-        s_subscriptions[rc.subId].balance -= payment;
-        s_withdrawableTokens += payment;
-      }
+      uint64 reqCount = s_subscriptions[rc.subId].reqCount;
+      s_subscriptions[rc.subId].reqCount = reqCount + 1;
```

Comment:
Shall we replace these two lines with the line below, since the local variable reqCount is not used anywhere else

> s_subscriptions[rc.subId].reqCount = s_subscriptions[rc.subId].reqCount + 1;